### PR TITLE
feat: make user first and last name optional fields

### PR DESCRIPTION
**Pull request by Alexei-Barnes**
This change partially completes #4386 by making the First Name and Last Name fields optional.

It completes the following acceptance criteria:

- [x] I can register with only a username
- [x] If I have set only a username in the ZITADEL console my username is shown instead of Firstname, Lastname
- [x] My Display Name shows the Firstname, Lastname if filled, if not it shows my username
- [x] We use the display name in emails instead of First and Lastname

It does not address the following acceptance criteria:

- [ ] If I do not have an email address on my user a password reset is not possible and shows an error
- [ ] If I don't have an email, the email setup or email verification screen is not shown in the login flow
- [ ] I am able to login if my user has set no email
- [ ] I am able to login if my user has set an email

This bites off the easiest chunk of this issue, since there is still some figuring out to do around user email address.

### CODE REVIEW

File: e2e/docker-compose.yaml [hunk 1]
```diff
@@ -39,7 +39,7 @@ services:
     user: '$UID'
     volumes:
       - .:/e2e
-    command: 'sh -c "npm ci --omit=dev && npm run lint && npx wait-on http://localhost:8080/debug/ready"'
+    command: 'sh -c "npm ci --omit=dev && npm run lint && npx wait-on -v -l http://host.docker.internal:8080/debug/ready"'
```
Comment (Alexei-Barnes) [inline e2e/docker-compose.yaml:42]: This should fix WSL support for this compose file - without this npx wait-on never finishes for me.

**Comment by Alexei-Barnes** [comment]
I suggest we should create a feature branch in the ziradel repo, and re-target this PR to that branch, since this doesn't complete all AC.

**Comment by hifabienne** [comment]
Hei @Alexei-Barnes 
Thank for the PR. 
Just one thing that comes to my mind is, if we merge this to main some of the existing customers will probably have a problem, because they assume that each user has a first- and a lastname.
We will probably have to implement an additional policy/settings where you can configure which fields are required and which are optional.

**Comment by Alexei-Barnes** [comment]
I did some testing on this today and found that this doesn't finish the problem of making first & last name optional - during registration from an IdP that didn't provide these details, the application got confused and produced the error `Errors.User.NotHuman`, and then `Errors.ExternalIDP.NoExternalUserData`. I'll need to investigate both of these issues.

**Comment by Alexei-Barnes** [comment]
I've managed to shift off other priorities and back to this, so hopefully I can fix those problems today.

**Comment by Alexei-Barnes** [comment]
OK the "NotHuman" error is simple;

`internal/user/repository/view/model/user.go:130`
```go
func (h *HumanView) IsZero() bool {
	return h == nil || h.FirstName == ""
}
```

will return true when called by

`internal/auth/repository/eventsourcing/eventstore/auth_request.go:662`
```go
	// if there's an active (human) user, let's use it
	if user != nil && !user.HumanView.IsZero() && domain.UserState(user.State).NotDisabled() {
		request.SetUserInfo(user.ID, loginName, user.PreferredLoginName, "", "", user.ResourceOwner)
		return nil
	}
```

and conclude that this user can't be found.

**Comment by Alexei-Barnes** [comment]
OK fixing that fixed both of the problems I identified, I managed to register a user via AWS Cognito with no first or last name configured, then use that to login.

**Comment by Alexei-Barnes** [comment]
I've noticed another problem with this change, which is that some pages show a robot avatar for the user, while others do not. I'll need to figure out why and update the UI to ensure the user is not considered a bot.

**Comment by hifabienne** [comment]
I have created a new user story regarding the topic about setting the required fields, before we can merge this, we should implement that: https://github.com/zitadel/zitadel/issues/5333

**Comment by muhlemmer** [comment]
As we are currently re-writing user management as [user schema](https://github.com/zitadel/zitadel/issues/6433), this PR seems to have become obsolete. Apologies for not merging your code. But know your effort was not in vain, it sparked a heavy and long-running debate which will result in a better zitadel that allows more flexibility in user management,

Thanks for the effort and hope seeing you around!

**Pull request closed without merging**